src/bootman/sysconfig.c

```diff
@@ -86,16 +86,67 @@ int cbm_get_fstype(const char *boot_device)
         return ret;
 }
 
+static void cmb_inspect_root_native(SystemConfig *c, char *realp) {
+        bool native_uefi = false;
+        autofree(char) *fw_path;
+        char *boot = NULL;
+
+        /* typically /sys, but we forcibly fail this with our tests */
+        fw_path = string_printf("%s/firmware/efi", cbm_system_get_sysfs_path());
+        native_uefi = nc_file_exists(fw_path);
+
+        /* Try to find the system ESP */
+        if (native_uefi && !getenv("CBM_FORCE_LEGACY")) {
+                boot = get_boot_device();
+                c->wanted_boot_mask |= BOOTLOADER_CAP_UEFI;
+
+                if (boot) {
+                        c->boot_device = boot;
+                        c->wanted_boot_mask |= BOOTLOADER_CAP_GPT;
+                        LOG_INFO("Discovered UEFI ESP: %s", boot);
+                }
+        } else {
+                /* Find legacy relative to root, on GPT */
+                boot = get_legacy_boot_device(realp);
+                c->wanted_boot_mask |= BOOTLOADER_CAP_LEGACY;
+
+                if (boot) {
+                        c->boot_device = boot;
+                        c->wanted_boot_mask |= BOOTLOADER_CAP_GPT;
+                        LOG_INFO("Discovered legacy boot device: %s", boot);
+                }
+        }
+}
+
+static void cmb_inspect_root_image(SystemConfig *c, char *realp) {
+        char *boot = NULL;
+
+        if (getenv("CBM_FORCE_LEGACY")) {
```

Comment:
So I think this breaks the extlinux method of image creation if clr-installer always sets this env. Is that the intent here? Also since we set the mbr boot flag for our efi installs I think you need to try and run get_boot_device first as we want to ensure the UEFI flag is set as we prefer that. Then if boot isn't found you should do the scan for legacy I think.

Reply:
Ah I think I misread the clr-installer change. So other than the extlinux concern it seems like this may fail to create an EFI image when the system creating the image doesn't have EFI support (as get_boot_device call expects the sysfs efi path). Previously the code would set BOOTLOADER_CAP_UEFI in the bootmask under those condititions.

Reply:
Yep, working on to fix the test break I saw the situations you mentioned, you're right, I'm changing the code to comply with that.

Basically the approach I'm assuming with that is:
+ check both for legacy and uefi boot
  + if both are "available" uefi has higher precedence
  + set legacy if it's the only "available"
  + set uefi if it's the only "available"
+ set legacy if CBM_FORCE_LEGACY is set

Reply:
Great!